### PR TITLE
docs: v0.4 java sdk

### PR DESCRIPTION
**Pull request by fengjiachun**
# Checklist

- [ ] Please ensure that the content in `summary.yml` matches the current document structure when you changed the document structure.

### CODE REVIEW

File: docs/en/v0.4/user-guide/write-data/sdk-libraries/java.md [hunk 1]
```diff
@@ -6,35 +6,65 @@
...
 
-After calling the `write` method on our completed `WriteRows`, a future will be returned which allows us to obtain write results through its callback function.
+After calling the `write` method on our completed `WriteRows`, a future will be returned which allows us to obtain write results through its callback function. You can also obtain the result synchronously by using `future.get()`.
+
+If you need to insert a large amount of data, consider using `StreamWriter` to enhance write performance. `StreamWriter` persistently writes data to the database using a `stream`. Once the writing is complete, you should invoke the `completed` method to close the `stream` and await the database's confirmation of write success. (Tip: Even though each `WriteRows` can only contain data from a single table, a `StreamWrite` instance can write multiple `WriteRows` simultaneously. This means that `StreamWrite` supports multi-table data import.)
```
Comment (nicecui) [inline docs/en/v0.4/user-guide/write-data/sdk-libraries/java.md:67]:
```suggestion
If you need to insert a large amount of data, consider using `StreamWriter` to enhance write performance. `StreamWriter` persistently writes data to the database using a `stream`. Once the writing is complete, you should invoke the `completed` method to close the `stream` and await the database's confirmation of write success. 

:::tip
Even though each `WriteRows` can only contain data from a single table, a `StreamWrite` instance can write multiple `WriteRows` simultaneously. This means that `StreamWrite` supports multi-table data import.
:::
```

File: docs/zh/v0.4/user-guide/query-data/sdk-libraries/java.md [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 ```java
 QueryRequest request = QueryRequest.newBuilder()
-    .exprType(SelectExprType.Sql) // Currently, only SQL is supported, and more query methods will be supported in the future
+    .exprType(SelectExprType.Sql) // It also supports PromQL query language (SelectExprType.Promql). Please refer to [promql](../promql.md).
```
Comment (nicecui) [inline docs/zh/v0.4/user-guide/query-data/sdk-libraries/java.md:9]: It is same as before, maybe we can move the link outside the java code block.

File: docs/zh/v0.4/user-guide/write-data/sdk-libraries/java.md [hunk 3]
```diff
@@ -6,33 +6,63 @@
...
+
+LOG.info("Write result: {}", result);
 ```
 
 在写入数据之前，我们需要创建一个 `TableSchema` 对象然后使用它来构造 `WriteRows` 对象。由于 `TableSchema` 可以被重复使用，可以将它缓存起来防止重复构建。
 
-当 `WriteRows` 被创建完成后，数据可以被添加到其中，但是此时这些数据仅仅存储在客户端内存中，还没有被发送到服务器。为了高效地插入多行数据，我们可以使用批量插入。在所有需要插入的数据都被添加到 `WriteRows` 后，调用它的 `finish` 方法来表示没有新数据要再被添加了，然后调用 DB 实例的 `write` 方法将数据发送到数据库中。在写入数据后，`Future` 对象会被返回，通过它的回调函数可以获取写入结果。
+当 `WriteRows` 被创建完成后，数据可以被添加到其中，但是此时这些数据仅仅存储在客户端内存中，还没有被发送到服务器。为了高效地插入多行数据，我们可以使用批量插入。在所有需要插入的数据都被添加到 `WriteRows` 后，调用它的 `finish` 方法来表示没有新数据要再被添加了，然后调用 DB 实例的 `write` 方法将数据发送到数据库中。在写入数据后，`Future` 对象会被返回，通过它的回调函数可以获取写入结果，也可以通过 `future.get()` 来同步的获得结果。
+
+如果你需要插入大量数据，可以考虑使用 `StreamWriter` 来提高写入性能。`StreamWriter` 会持续地将数据通过 `stream` 写入数据库。写入完成后，你需要调用 `completed` 方法关闭 `stream` 并等待数据库返回写入是否成功的信息。 （提示：尽管每个 `WriteRows` 只能包含一个表的数据，但一个 `StreamWrite` 实例可以分批写入多个 `WriteRows`，这意味着 `StreamWrite` 支持多表数据的导入。）
```
Comment (nicecui) [inline docs/zh/v0.4/user-guide/write-data/sdk-libraries/java.md:65]:
```suggestion
如果你需要插入大量数据，可以考虑使用 `StreamWriter` 来提高写入性能。`StreamWriter` 会持续地将数据通过 `stream` 写入数据库。写入完成后，你需要调用 `completed` 方法关闭 `stream` 并等待数据库返回写入是否成功的信息。 

:::tip
尽管每个 `WriteRows` 只能包含一个表的数据，但一个 `StreamWrite` 实例可以分批写入多个 `WriteRows`，这意味着 `StreamWrite` 支持多表数据的导入。
:::
```

File: docs/zh/v0.4/user-guide/query-data/sdk-libraries/java.md (hunk 2 above)
Comment (nicecui) [inline docs/zh/v0.4/user-guide/query-data/sdk-libraries/java.md:9]:
```suggestion
    .exprType(SelectExprType.Sql) // It also supports PromQL query language (SelectExprType.Promql)
```

**Review by nicecui — APPROVED** [review]

**Review by xtang — APPROVED** [review]

**Pull request merged**